rbx_reflection_database/src/lib.rs

```diff
@@ -14,10 +14,32 @@ pub fn get() -> &'static ReflectionDatabase<'static> {
 
 #[cfg(test)]
 mod test {
+    use rbx_reflection::ClassDescriptor;
+
     use super::*;
 
     #[test]
     fn smoke_test() {
         let _database = get();
     }
+
+    #[test]
+    fn superclasses_iter_test() {
+        let database = get();
+        let part_class_descriptor = database.classes.get("Part");
+        let mut iter = database.superclasses_iter(part_class_descriptor.unwrap());
+        fn class_descriptor_eq(lhs: Option<&ClassDescriptor>, rhs: Option<&ClassDescriptor>) {
+            assert!(match (lhs, rhs) {
+                (Some(lhs), Some(rhs)) => std::ptr::addr_eq(lhs, rhs),
+                (None, None) => true,
+                _ => false,
+            })
```

Comment:
I don't support using pointer equality for this; it feels... unnecessary.

```suggestion
            match (lhs, rhs) {
                (Some(lhs), Some(rhs)) => assert_eq!(lhs.name, rhs.name),
                (None, None) => true,
                _ => panic!("{:?} != {:?}", lhs, rhs),
            }
```

Reply:
I figured there shouldn't be duplicate objects, but I suppose that's not part of this test.  I made some changes and worked in the print expression.